community/kernel/src/main/java/org/neo4j/kernel/recovery/LatestCheckPointFinder.java

```diff
@@ -91,9 +93,28 @@ public LatestCheckPoint find( long fromVersionBackwards ) throws IOException
 
             if ( latestCheckPoint != null )
             {
+                // Is the latest start entry in this log file version later than what the latest check point targets?
+                LogPosition target = latestCheckPoint.getLogPosition();
                 boolean commitsAfterCheckPoint = latestStartEntry != null &&
-                        latestStartEntry.getStartPosition().compareTo( latestCheckPoint.getLogPosition() ) >= 0;
-                return new LatestCheckPoint( latestCheckPoint, commitsAfterCheckPoint , oldestVersionFound );
+                        latestStartEntry.getStartPosition().compareTo( target ) >= 0;
+                if ( !commitsAfterCheckPoint )
+                {
+                    if ( target.getLogVersion() < version )
+                    {
+                        // This check point entry targets a previous log file.
+                        // Go there and see if there's a transaction. Reader is capped to that log version.
+                        commitsAfterCheckPoint = extractFirstTxIdAfterPosition( target, target.getLogVersion() ) !=
```

Comment:
Transactions are written fully in one or the other tx log, never cut off in the middle. A check point can be made to target the log position either before or after tx1 and it doesn't matter, because `extractFirstTxIdAfterPosition` will go to the target log position and find the first commit entry, crossing log file boundaries if it needs to.